### PR TITLE
feat(examples): adds example for running jest with typescript

### PR DESCRIPTION
**Pull request by fredrikredflag**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] Other... Please describe: example

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

Adds new example for using jest with typescript

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

closes: #2135

### CODE REVIEW

**Comment by fredrikredflag** [comment]
@googlebot I signed it!

**Comment by fredrikredflag** [comment]
For reviewers: I had my other email configured on my laptop, hence the problem with google-cla.

File: examples/BUILD.bazel [hunk 1]
```diff
@@ -113,6 +113,10 @@ example_integration_test(
 
 example_integration_test(
     name = "examples_jest",
+    bazel_commands = ["build ..."],
```
Comment (alexeagle) [inline examples/BUILD.bazel:116]: this looks wrong, we won't run `bazel test` on the example anymore, so how do we know if it works?

File: examples/jest/ts/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (alexeagle) [inline examples/jest/ts/tsconfig.json:1]: maybe this file should go one directory up? is that more typical of how it's used?
then you wouldn't need the `alias` rule in the /BUILD.bazel

**Review by alexeagle — COMMENTED** [review]
thanks!

File: examples/BUILD.bazel (hunk 1 above)
Comment (fredrikredflag) [reply examples/BUILD.bazel:116]: Oh, copy paste :)

Didn't really understand how the integration tests work

File: examples/jest/ts/tsconfig.json (hunk 2 above)
Comment (fredrikredflag) [reply examples/jest/ts/tsconfig.json:1]: fixing

**Review by alexeagle — APPROVED** [review]
thanks!

**Pull request merged**